reps/serve_pipeline.md

```diff
@@ -0,0 +1,139 @@
+
+## Summary - Serve Pipeline
+### General Motivation
+Production machine learning serving pipelines are getting longer and wider. They often consist of multiple, or even tens of models collectively making a final prediction, such as image / video content classification and tagging, fraud detection pipeline with multiple policies and models, multi-stage ranking and recommendation, etc.
+
+Meanwhile, the size of a model is also growing beyond the memory limit of a single machine due to the exponentially growing number of parameters, such as GPT-3, sparse feature embeddings in recsys models such that the ability to do disaggregated and distributed inference is desirable and future proof.
+
+We want to leverage the programmable and general purpose distributed computing ability of Ray, double down on its unique strengths (scheduling, communication and shared memory) to facilitate authoring, orchestrating, scaling and deployment of complex serving pipelines under one set of DAG API, so a user can program & test multiple models or multiple shards of a single large model dynamically, deploy to production at scale, and upgrade individually.
+#### Key requirements:
+- Provide the ability to author a DAG of serve nodes to form a complex inference graph.
+- Pipeline authoring experience should be fully python programmable with support for dynamic selection, control flows, user business logic, etc.
+- DAG can be instantiated and locally executed using tasks and actors API
+- DAG can be deployed via declarative and idempotent API, individual nodes can be reconfigured and scaled indepenently. 
+
+### Should this change be within `ray` or outside?
+main `ray` project. Changes are made to Ray Core and Ray Serve level.
+
+## Stewardship
+### Required Reviewers
+The proposal will be open to the public, but please suggest a few experience Ray contributors in this technical domain whose comments will help this proposal. Ideally, the list should include Ray committers.
+
+@ericl, @edoakes, @simon-mo, @jiaodong
+
+### Shepherd of the Proposal (should be a senior committer)
+To make the review process more productive, the owner of each proposal should identify a **shepherd** (should be a senior Ray committer). The shepherd is responsible for working with the owner and making sure the proposal is in good shape (with necessary information) before marking it as ready for broader review.
+
+@ericl
```

Comment:
@zhe-thoughts should the shepherd just be the person reviewing / merging the PR into this repo? If so, we don't need to list it explicitly in the REP.

Reply:
@ericl Yes I think the workflow can be that the proposer designate the shepherd on the `Reviewers` field in the PR, and the shepherd should be the one to merge it. But how do people find the info once the REP is a committed .md file? As long as they can find out who's the shepherd, I'm OK with it

Reply:
Could always look through the merged PRs list (i.e., this thread).